api/repositories/process_repository.go

```diff
@@ -241,6 +241,41 @@ func (r *ProcessRepo) GetProcessByAppTypeAndSpace(ctx context.Context, authInfo
 	return returnProcess(matches)
 }
 
+func (r *ProcessRepo) GetAppRevisionForProcess(ctx context.Context, authInfo authorization.Info, appGUID, spaceGUID string) (string, error) {
+	var appRevision string
+	userClient, err := r.clientFactory.BuildClient(authInfo)
+	if err != nil {
+		return appRevision, fmt.Errorf("get-apprevision-for-process: failed to build user k8s client: %w", err)
+	}
+
+	appList := new(korifiv1alpha1.CFAppList)
+	err = userClient.List(ctx, appList, client.InNamespace(spaceGUID))
```

Comment:
No need to list apps here, we know the guid so we can just get it in the same way the [app repo](https://github.com/cloudfoundry/korifi/blob/e5f58770befffd0520e5f2fe52f574ce55ba4f69/api/repositories/app_repository.go#L204) does. If we do this, we won't need to manually filter through the list later on.

Reply:
aah, got it now!